parquet/src/test/java/org/apache/iceberg/parquet/TestParquetSchemaUtil.java

```diff
@@ -464,4 +588,103 @@ private Type map(Integer id, String name, Repetition repetition, Type keyType, T
     }
     return builder.named(name);
   }
+
+  @Test
+  public void testPruneColumns1() {
+    MessageType messageType = ParquetSchemaUtil.convert(ICEBERG_SCHEMA, "schema");
+    Type visit =
+        TypeWithSchemaVisitor.visit(
+            ICEBERG_SCHEMA.asStruct(), messageType, new PruneColumns(ImmutableSet.of(6, 7, 8)));
+    String expectedType =
```

Comment:
Why are we using the String? Could we just build up a Schema object with the correct layout?